xarray/core/formatting_html.py

```diff
@@ -0,0 +1,313 @@
+             .decode('utf8'))
+
+
+def format_dims(dims, coord_names):
+    if not dims:
+        return ''
+
+    dim_css_map = {k: " class='xr-has-index'" if k in coord_names else ''
+                   for k, v in dims.items()}
+
+    dims_li = "".join("<li><span{cssclass_idx}>{name}</span>: {size}</li>"
+                      .format(cssclass_idx=dim_css_map[k], name=k, size=v)
+                      for k, v in dims.items())
+
+    return "<ul class='xr-dim-list'>{}</ul>".format(dims_li)
+
+
+def format_values_preview(array, max_char=35):
+    pprint_str = format_array_flat(array, max_char)
+
+    return "".join("<span>{} </span>".format(s)
+                   for s in pprint_str.split())
+
+
+def summarize_attrs(attrs):
+    attrs_li = "".join("<li>{} : {}</li>".format(k, v)
+                       for k, v in attrs.items())
+
+    return "<ul class='xr-attrs'>{}</ul>".format(attrs_li)
+
+
+def _icon(icon_name):
+    # icon_name should be defined in xarray/static/html/icon-svg-inline.html
+    return ("<svg class='icon {0}'>"
+            "<use xlink:href='#{0}'>"
+            "</use>"
+            "</svg>"
+            .format(icon_name))
+
+
+def _summarize_coord_multiindex(name, coord, d):
+    d['dtype'] = 'MultiIndex'
+    d['preview'] = '(' +  ', '.join(l for l in coord.level_names) + ')'
+    return summarize_variable(name, coord, d)
+
+
+def summarize_coord(name, var):
+    d = {}
+    is_index = name in var.dims
+    d['cssclass_idx'] = " class='xr-has-index'" if is_index else ""
+    if is_index:
+        coord = var.variable.to_index_variable()
+        if coord.level_names is not None:
+            coords = {}
+            coords[name] = _summarize_coord_multiindex(name, coord, d)
+
+            for lname in coord.level_names:
+                var = coord.get_level_variable(lname)
+                coords[lname] = summarize_variable(lname, var)
+
+            return coords
+
+    return {name: summarize_variable(name, var.variable, d)}
+
+
+def summarize_coords(variables):
+    coords = {}
+    for k, v in variables.items():
+        coords.update(**summarize_coord(k, v))
+
+    vars_li = "".join("<li class='xr-var-item'>{}</li>"
+                     .format(v) for v in coords.values())
+
+    return "<ul class='xr-var-list'>{}</ul>".format(vars_li)
+
+
+def summarize_variable(name, var, d=None):
+    if d is None:
+        d = {'cssclass_idx': ""}
+
+    d['dims_str'] = '(' +  ', '.join(dim for dim in var.dims) + ')'
+    d['name'] = name
+    d['dtype'] = d.get('dtype', var.dtype)
+
+    # "unique" ids required to expand/collapse subsections
+    d['attrs_id'] = 'attrs-' + str(uuid.uuid4())
+    d['data_id'] = 'data-' + str(uuid.uuid4())
+
+    if len(var.attrs):
+        d['disabled'] = ''
+        d['attrs'] = summarize_attrs(var.attrs)
+    else:
+        d['disabled'] = 'disabled'
+        d['attrs'] = ''
+
+    # TODO: no value preview if not in memory
+    d['preview'] = d.get('preview', format_values_preview(var))
```

Comment:
We should probably implement this TODO before merging. Otherwise printing the repr() will probably cause computers to crash when using large datasets with dask-distributed.

Reply:
Also: this could be a nice place to recursively check for an `_html_repr_` attribute, e.g., on dask arrays!

Reply:
oh I didn't look closely and read that TODO as a DONE. Yes I agree that we should make sure the data are loaded.

Reply:
Yes, would be amazing to see dask's new fancy array html repr here!

Reply:
I would suggest using `inline_variable_array_repr` from `formatting.py`.

Reply:
> I would suggest using inline_variable_array_repr from formatting.py

The issue with just reusing `inline_variable_array_repr` here is that it would be then more tricky to use html templates to tweak the rendering of the values (e.g., encapsulate each value in its own `span` with custom padding). See [my comment below](https://github.com/pydata/xarray/pull/3425#issuecomment-544839504).

Reply:
Mmm there is apparently no custom padding in the CSS, so maybe using `span` for each value is not really needed?

Here is how it looks without alternating grey coloring:

<img width="416" alt="Screenshot 2019-10-22 at 09 52 26" src="https://user-images.githubusercontent.com/4160723/67267024-c7ccfe80-f4b1-11e9-97b4-680a6a631462.png">

What do you think about the spacing between the values? Is it enough?

Reply:
I think it's enough.